src/connection.js

```diff
@@ -353,6 +371,29 @@ class Connection extends EventEmitter {
       if (config.options.useUTC != undefined) {
         this.config.options.useUTC = config.options.useUTC;
       }
+
+      // Whenever authentication property is used to specify an authentication method,
+      // the client will request encryption (the default value of the Encrypt property will be true)
+      // and it will validate the server certificate (regardless of the encryption setting), unless TrustServerCertificate = true
+      if (config.options.authentication != undefined) {
+        // check for valid options
+        if (!(config.options.authentication.toUpperCase() === this.fedAuthInfo.ValidFedAuthEnum.SqlPassword ||
+        config.options.authentication.toUpperCase() === this.fedAuthInfo.ValidFedAuthEnum.ActiveDirectoryPassword)) {
+          throw new Error('An invalid authentication method is specified');
+        }
+        if (this.config.options.tdsVersion < '7_4') {
+          throw new Error(`Azure Active Directory authentication is not supported in the TDS version ${this.config.options.tdsVersion}`);
+        }
+        this.config.options.encrypt = true;
+        this.fedAuthInfo.method = config.options.authentication;
+        if (config.options.trustServerCertificate != 'true') {
+          this.config.options.trustServerCertificate = false;
```

Comment:
`this.config.options.trustServerCertificate` is already has a boolean value (set [here](https://github.com/tediousjs/tedious/pull/661/files#diff-875da84119066ec244742e95d1209f0cR363)), can you remove this validation?

Reply:
it could be set to anything (for example it could be 'abcd') up there, and I wanna make sure unless it is explicitly set to true by user, it is set to false.

Reply:
You're right! For most of the boolean options, error is thrown if value is not boolean (like [here](https://github.com/tediousjs/tedious/pull/661/files#diff-875da84119066ec244742e95d1209f0cR274)), looks like `config.options.trustServerCertificate` was missed. Can you do the same for `trustServerCertificate` too? It would be more generic.

Reply:
Yeah, we should catch this and require it to be a `boolean` if it is set. As this change is orthogonal to active directory auth, could you do this in a separate pull request?

Reply:
@v-suhame @arthurschreiber sent  PR #692 for checking the boolean config types